src/microbe_stage/ProcessSystem.cs

```diff
@@ -377,81 +381,115 @@ private void ProcessNode(IProcessable processor, float delta, float inverseDelta
         }
 
         if (environmentModifier <= MathUtils.EPSILON)
-        {
             canDoProcess = false;
-        }
 
+        // Compute spaceConstraintModifier before updating the final use and input amounts
         foreach (var entry in processData.Inputs)
         {
             if (entry.Key.IsEnvironmental)
                 continue;
 
-            var inputRemoved = entry.Value * process.Rate * delta;
-
-            currentProcessStatistics?.AddInputAmount(entry.Key, inputRemoved * inverseDelta);
+            var inputRemoved = entry.Value * process.Rate * delta * environmentModifier * spaceConstraintModifier;
 
             // currentProcessStatistics?.AddInputAmount(entry.Key, 0);
+            // We don't multiply by delta here because we report the per-second values anyway. In the actual process
+            // output numbers (computed after testing the speed), we need to multiply by inverse delta
+            currentProcessStatistics?.AddInputAmount(entry.Key, entry.Value * process.Rate * environmentModifier);
```

Comment:
You could divide inputRemoved by the delta here and avoid the formulae going out of sync.

Reply:
It's just a few lines away. And anyway there's 4 closely related formulas in this method already, so I thought that the performance win of avoiding a divide is better here.

Reply:
An alternative option would be to just multiply everything BUT the delta first, then calculate the input removed afterwards. That saves you even more performance.

Reply:
That sounds like a good idea. I'll do that once I'm done working on the feature for ThriveDevCenter that I'm currently working on.

Reply:
Done